src/evaluation/positioning.h

```diff
@@ -5,136 +5,210 @@
 #include <array>
 #include <cstdint>
 
+#include "evaluation/game_phase.h"
 #include "evaluation/position_tables.h"
 #include "helpers/bit_operations.h"
+#include "helpers/game_phases.h"
 #include "movegen/bishops.h"
 #include "movegen/kings.h"
+#include "movegen/knights.h"
 #include "movegen/rooks.h"
 
 namespace evaluation {
 
 namespace {
 
-constexpr int32_t s_doublePawnPenalty { -10 };
-constexpr int32_t s_isolatedPawnPenalty { -10 };
-constexpr auto s_passedPawnBonus = std::to_array<int32_t>({ 0, 10, 30, 50, 75, 100, 150, 200 });
-constexpr int32_t s_semiOpenFileScore { 10 };
-constexpr int32_t s_openFileScore { 15 };
+/* TODO: find a way to tune these parameters in an automated way - currently they're just guesses
```

Comment:
Does this point to the possibility of tuning more generally throughout the engine? AFAI can see the only parameters we're manually obtaining are the magic numbers for sliding pieces.

Reply:
Yes exactly! We need to build a tuner that can evaluate positions, based on a known dataset, and then optimize the parameters based on that. Something else todo 🙃